src/components/Menu/index.tsx

```diff
@@ -1,158 +1,107 @@
-    ${endStyles};
-    ${bottomStyles(theme)};
-  `,
-  'bottom-start': (theme: Theme) => css`
-    ${startStyles};
-    ${bottomStyles(theme)};
-  `,
-  top: (theme: Theme) => css`
-    ${centerStyles};
-    ${topStyles(theme)};
-  `,
-  'top-end': (theme: Theme) => css`
-    ${endStyles};
-    ${topStyles(theme)};
-  `,
-  'top-start': (theme: Theme) => css`
-    ${startStyles};
-    ${topStyles(theme)};
-  `,
-} as const
-
-type ArrowPlacement = keyof typeof arrowPlacementStyles
+type ArrowPlacement =
+  | 'bottom'
+  | 'bottom-end'
+  | 'bottom-start'
+  | 'top'
+  | 'top-end'
+  | 'top-start'
 type MenuListProps = {
   align: AlignStyle
   hasArrow: boolean
+  placement: ArrowPlacement
 }
-export const arrowPlacements = Object.keys(
-  arrowPlacementStyles,
-) as ArrowPlacement[]
 
 type AlignStyle = {
   left?: string | null
   right?: string | null
 }
 
+interface ChildrenProps {
+  placement: ComponentProps<typeof Popper>['placement']
+  toggle: () => void
+  visible: boolean
+}
+
+type MenuProps = Omit<ComponentProps<typeof Popper>, 'children'> & {
+  align?: AlignStyle
+  ariaLabel?: string
+  placement?: ArrowPlacement
+  children?: ((props: ChildrenProps) => ReactNode) | ReactNode
+  className?: string
+}
+
+type MenuType = FunctionComponent<MenuProps> & { Item: typeof Item }
+
 const MenuList = styled.div<MenuListProps>`
   &:after,
   &:before {
     left: ${({ align }) => align.left};
-    right: ${({ align }) => align.left};
-  }
-  background-color: ${({ theme }) => theme.colors.neutral.backgroundWeak};
-  display: flex;
-  flex-direction: column;
-  text-align: center;
-  justify-content: center;
-  color: ${({ theme }) => theme.colors.neutral.text};
-  border-radius: 4px;
-  position: relative;
+    right: ${({ align }) => align.right};
 
-  &:after,
-  &:before {
     border: solid transparent;
     border-width: 9px;
     content: ' ';
     height: 0;
     width: 0;
     position: absolute;
     pointer-events: none;
+    bottom: ${({ placement }) => (placement.includes('bottom') ? '100%' : '')};
+    top: ${({ placement }) => (placement.includes('top') ? '100%' : '')};
+    left: ${({ placement }) =>
+      !placement.includes('start') && !placement.includes('end') ? '50%' : ''};
+    transform: ${({ placement }) =>
+      !placement.includes('start') && !placement.includes('end')
+        ? 'translateX(-50%)'
+        : ''};
   }
 
   &:after {
     border-color: transparent;
+    border-bottom-color: ${({ theme, placement }) =>
+      placement.includes('bottom') && theme.colors.primary.borderStrong};
+    border-top-color: ${({ theme, placement }) =>
+      placement.includes('top') && theme.colors.primary.borderStrong};
+    margin-left: ${({ placement }) => placement.includes('end') && '-9px'};
+    right: ${({ placement }) => placement.includes('end') && '24px'};
+    margin-right: ${({ placement }) => placement.includes('start') && '-9px'};
+    left: ${({ placement }) => placement.includes('start') && '24px'};
```

Comment:
Ah euh, this seems a lot more complicated than before, we could have the same behaviour as before, but move the computing of the variant inside the styled :)